### PR TITLE
Symfony 2.7 compatibility

### PR DESCRIPTION
**Pull request by malarzm**

### CODE REVIEW

**Comment by malarzm** [comment]
Ok, I went through Symfony's 2.8 changelog and it seems these changes should be enough... @stof @alcaeus do you recall anything else that needs to be worked around?

**Comment by alcaeus** [comment]
@malarzm Can't think of any - IIRC the type stuff was the only thing changed in 2.8.

File: Form/DoctrineMongoDBTypeGuesser.php [hunk 1]
```diff
@@ -39,9 +40,12 @@ class DoctrineMongoDBTypeGuesser implements FormTypeGuesserInterface
 
     private $cache = array();
 
+    private $typeFQCN;
+
     public function __construct(ManagerRegistry $registry)
     {
         $this->registry = $registry;
+        $this->typeFQCN = version_compare(Kernel::VERSION, '2.8.0', '>=');
```
Comment (stof) [inline Form/DoctrineMongoDBTypeGuesser.php]: this should not detect the kernel version but the version of the Form component. It is perfectly valid to install the Form 2.8 component with HttpKernel 2.7.

Use ``method_exists('\Symfony\Component\Form\AbstractType', 'getBlockPrefix')`` to detect the new form component

**Comment by stof** [comment]
IIRC, the decision to require 2.8+ was made to simplify the need to handle BC layers (and 2.8 is the current LTS)

File: Form/DoctrineMongoDBTypeGuesser.php (hunk 1 above)
Comment (malarzm) [inline Form/DoctrineMongoDBTypeGuesser.php]: will fix later, thanks for the tip :+1:

**Comment by malarzm** [comment]
Personally if we'll get away with only these changes I'll be fine with supporting 2.7

**Pull request merged**